packages/apps/transfer/src/services/transfer/coin-transfer.ts

```diff
@@ -77,3 +77,94 @@ export async function transferCreate(
   await pactCommand.send(generateApiHost(networkId, chainId));
   return pactCommand;
 }
+
+export async function safeTransferCreate(
+  fromAccount: string,
+  toAccount: string,
+  amount: string,
+  fromPrivateKey: string,
+  toPrivateKey: string,
+  chainId: ChainId,
+  networkId: ChainwebNetworkId,
+): Promise<PactCommand> {
+  if (isNaN(Number(amount))) {
+    throw new Error('Amount must be a number');
+  }
+
+  // sender need to send extra coins to the receiver which will be returned by the receiver
+  const amountWithExtra = Number(amount) + 0.000000000001;
+
+  const pactCommand = new PactCommand();
+  pactCommand.code = `(coin.transfer-create "${fromAccount}" "${toAccount}" (read-keyset "ks") ${convertDecimal(
+    amountWithExtra.toString(),
+  )})(coin.transfer "${toAccount}" "${fromAccount}" ${convertDecimal(
+    '0.000000000001',
+  )})`;
+
+  pactCommand
+    .addCap('coin.GAS', onlyKey(fromAccount))
+    .addCap(
+      'coin.TRANSFER',
+      onlyKey(fromAccount),
+      fromAccount,
+      toAccount,
+      Number(amountWithExtra),
+    )
+    .addCap(
+      'coin.TRANSFER',
+      onlyKey(toAccount),
+      toAccount,
+      fromAccount,
+      Number('0.1'),
+    )
+    .addData({ ks: { pred: 'keys-all', keys: [onlyKey(toAccount)] } })
+    .setMeta(
+      {
+        gasLimit: gasLimit,
+        gasPrice: gasPrice,
+        ttl: ttl,
+        chainId: chainId,
+        sender: fromAccount,
+      },
+      networkId,
+    );
+
+  pactCommand.createCommand();
+
+  if (pactCommand.cmd === undefined) {
+    throw new Error('Failed to create transaction');
+  }
+
+  const signatureSender = sign(pactCommand.cmd, {
+    secretKey: fromPrivateKey,
+    publicKey: onlyKey(fromAccount),
+  });
+
+  const signatureReceiver = sign(pactCommand.cmd, {
+    secretKey: toPrivateKey,
+    publicKey: onlyKey(toAccount),
+  });
+
+  if (
+    signatureSender.sig === undefined ||
+    signatureReceiver.sig === undefined
+  ) {
+    throw new Error('Failed to sign transaction');
+  }
```

Comment:
The sender and receiver could be signed in a separate process where the transaction from `createCommand()` could be sent to the other person via an async process. 
Does the feature allow for that use case?